Automattic/woocommerce-subscriptions-core
# Add JS validation to prevent users from creating free subscription products

Fixes https://github.com/Automattic/woocommerce-payments/issues/3309

## Description

As described in https://github.com/Automattic/woocommerce-payments/issues/3309, free subscription products are a feature of the plugin and shouldn't be available to libraries using subscriptions-core.

This PR adds JS validation to the edit product screen to prevent merchants from creating free subscription products.

<img width="991" alt="Screen Shot 2021-12-02 at 4 49 51 pm" src="https://user-images.githubusercontent.com/8490476/144372265-1ee461ce-2395-4e02-bdee-8905a29cb73f.png">

## How to test this PR

1. Activate this branch of subscriptions core on your local site.
2. With the WC Subscriptions plugin **deactivated**:
    1. Go to **Products > Add** new from your admin dashboard.
    2. Select a subscription product type (**Simple subscription** or **Variable subscription**).
    3. _Add variations if you are creating a variable subscription._
    4. Attempt to enter a price of $0.
    5. You should see the warning message (see screenshot).
    6. If you leave the price field (change focus), the price will be emptied. 
         - _**Note**: If saved, products with  empty prices are un-purchaseable._
3. With the WC Subscriptions plugin **activated**:
    1. Go to **Products > Add** new from your admin dashboard.
    2. Select a subscription product type (**Simple subscription** or **Variable subscription**).
    3. _Add variations if you are creating a variable subscription._
    4. Attempt to enter a price of $0.
    5. It should be allowed with no error message.

**Additional notes:**

- If you enter a price like 0.005 when your store only has 2 decimal places. The warning will also be displayed.
- The notice uses the store's currency format (see examples below). 

<img width="657.75" alt="Screen Shot 2021-12-02 at 5 07 48 pm" src="https://user-images.githubusercontent.com/8490476/144374536-8117e295-07aa-4085-8eca-574dec188274.png">


## Product impact


- [ ] Will this PR affect WooCommerce Subscriptions? yes/no/tbc, add issue ref - NA
    - https://github.com/Automattic/woocommerce-payments/issues/3309
